### PR TITLE
MRG: Refactor plot_alignment

### LINKED ISSUE #9715: BUG: plot_alignment EEG projection incorrect
See https://github.com/mne-tools/mne-python/pull/9585#issuecomment-910546309

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR refactors `plot_alignment()` following https://github.com/mne-tools/mne-python/pull/9689#issuecomment-911385945. It  fixes https://github.com/mne-tools/mne-python/issues/9542. It fixes #9715.

It's still work in progress.

**ToDO**
- [x]  _plot_head_surface
- [x] _plot_axes
- [x] _plot_head_shape_points
- [x] _plot_head_fiducials
- [x] _plot_mri_fiducials
- [x] _plot_forward

The functions should all receive an instance of `_Renderer` as first parameter and return a list of `vtkActor`s as first output.

### CODE REVIEW

**Comment by larsoner** [comment]
If possible could you also fix #9715 / https://github.com/mne-tools/mne-python/pull/9585#issuecomment-910546309 ?

**Comment by GuillaumeFavelier** [comment]
I updated the branch with `main` and I think I fixed it indirectly.

https://33649-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/forward/35_eeg_no_mri.html?highlight=eeg%20mri

**Review by agramfort — COMMENTED** [review]
so far so good !

@GuillaumeFavelier tell me when it's good enough from your end.

File: mne/viz/_3d.py [hunk 1]
```diff
@@ -986,9 +966,36 @@ def _plot_head_shape_points(renderer, info, to_cf_t):
...
+    green = (0.0, 1.0, 0.0)
+    blue = (0.0, 0.0, 1.0)
+    for ori, color in zip(range(fwd_nn.shape[1]), (red, green, blue)):
+        renderer.quiver3d(fwd_rr[:, 0],
+                          fwd_rr[:, 1],
+                          fwd_rr[:, 2],
+                          fwd_nn[:, ori, 0],
+                          fwd_nn[:, ori, 1],
+                          fwd_nn[:, ori, 2],
+                          color=color, mode='arrow', scale=1.5e-3)
```
Comment (agramfort) [inline mne/viz/_3d.py:998]: you should return something no?

Comment (GuillaumeFavelier) [reply mne/viz/_3d.py]: Done in fa90c93 !

**Comment by GuillaumeFavelier** [comment]
It's now ready on my end @agramfort, what do you think?

**Comment by larsoner** [comment]
Also @alexrockhill feel free to look if you have time

**Review by agramfort — APPROVED** [review]

**Comment by alexrockhill** [comment]
Didn't run it but looks like great fixes, thanks and sorry I didn't have the time to do the bug I made

**Pull request merged**